internal/resources/grafana/resource_report.go

```diff
@@ -571,3 +583,9 @@ func parseCustomReportInterval(i interface{}) (int, string, error) {
 
 	return number, unit, nil
 }
+
+func validateTimezone(i interface{}) error {
+	timezone := i.(string)
+	_, err := time.LoadLocation(timezone)
+	return err
+}
```

Comment:
```suggestion
func validateTimezone(i interface{}, path cty.Path) diag.Diagnostics {
	timezone := i.(string)
	_, err := time.LoadLocation(timezone)
	return diag.FromErr(err)
}
```

This will allow you to use it directly: `ValidateDiagFunc: validateTimezone`